ykpack/src/types.rs

```diff
@@ -516,46 +561,57 @@ impl Statement {
     /// Returns a vector of locals that this SIR statement uses.
     /// A definition is considered a use, so this returns a superset of what
     /// `maybe_defined_locals()` does.
+    /// FIXME above
     pub fn used_locals(&self) -> Vec<Local> {
         let mut ret = Vec::new();
 
         match self {
             Statement::Nop => (),
-            Statement::Assign(place, rval) => {
-                rval.push_used_locals(&mut ret);
-                place.push_used_locals(&mut ret);
+            Statement::Debug(_) => (),
+            Statement::MkRef(dest, src) => {
+                Self::maybe_push_local(&mut ret, dest.local());
+                Self::maybe_push_local(&mut ret, src.local());
             }
-            Statement::Enter(_target, args, dest, start_idx) => {
-                if let Some(dest) = dest {
-                    dest.push_used_locals(&mut ret);
-                }
-                for a in args {
-                    a.push_used_locals(&mut ret);
-                }
-                for idx in 0..args.len() {
-                    // + 1 to skip return value.
-                    ret.push(Local(start_idx + u32::try_from(idx).unwrap() + 1));
-                }
+            Statement::DynOffs {
+                dest, base, idx, ..
+            } => {
+                Self::maybe_push_local(&mut ret, dest.local());
+                Self::maybe_push_local(&mut ret, base.local());
+                Self::maybe_push_local(&mut ret, idx.local());
+            }
+            Statement::BinaryOp {
+                dest, opnd1, opnd2, ..
+            } => {
+                Self::maybe_push_local(&mut ret, opnd1.local());
+                Self::maybe_push_local(&mut ret, opnd2.local());
+                Self::maybe_push_local(&mut ret, dest.local());
+            }
+            Statement::IStore(dest, src) => {
+                Self::maybe_push_local(&mut ret, dest.local());
+                Self::maybe_push_local(&mut ret, src.local());
             }
-            Statement::Leave => (),
             Statement::StorageDead(_) => (),
             Statement::Call(_target, args, dest) => {
                 if let Some(dest) = dest {
-                    dest.push_used_locals(&mut ret);
+                    Self::maybe_push_local(&mut ret, dest.local());
                 }
                 for a in args {
-                    a.push_used_locals(&mut ret);
+                    Self::maybe_push_local(&mut ret, a.local());
                 }
             }
+            Statement::Cast(dest, src) => {
+                Self::maybe_push_local(&mut ret, dest.local());
+                Self::maybe_push_local(&mut ret, src.local());
+            }
             Statement::Unimplemented(_) => (),
         }
         ret
     }
 
-    /// Returns true if the statement may affect locals besides those appearing in the statement.
+    ///// Returns true if the statement may affect locals besides those appearing in the statement.
```

Comment:
Accidental change?

Reply:
Yes, where i have commented and uncommented the block. Thanks.